components/layout/css/matching.rs

```diff
@@ -289,6 +289,12 @@ impl StyleSharingCandidate {
             return false
         }
 
+        let mut matching_rules = vec![];
+        element.synthesize_presentational_hints_for_legacy_attributes(&mut matching_rules);
+        if !matching_rules.is_empty() {
+            return false;
+        }
```

Comment:
This is probably going to be too slow. `can_share_style_with` is called zillions of times. Can you avoid creating the vector?